src/Microsoft.Azure.WebJobs.Host/Executors/FunctionInstance.cs

```diff
@@ -51,6 +52,18 @@ public IFunctionInvoker Invoker
             get { return _invoker; }
         }
 
+        public object JobInstance
```

Comment:
The job instance was needed earlier in the pipeline for exception filters, so I'm moving to a create/cache model so other places in the pipeline can access the cached instance